src/app/InteractionModelEngine.cpp

```diff
@@ -985,7 +977,6 @@ void InteractionModelEngine::OnResponseTimeout(Messaging::ExchangeContext * ec)
                  ChipLogValueExchange(ec));
 }
 
-#if CHIP_CONFIG_ENABLE_READ_CLIENT
```

Comment:
Comment to serve as a "changes requested". I believe these actually calling into ReadHandler ends up pulling the entire ReadHandle code.

Reply:
Yes, exactly.  That is the problem: virtual functions and then everything gets pulled in.

The only way to avoid it is to not have any code in the build that touches the vtable.